### ISSUE TITLE
Importing theano before pymc3 results in ImportError, but only when using GPU

### ISSUE THREAD

**Issue by Spaak** [pymc-devs/pymc#1613]
This is quite an obscure bug. The following works fine when using a CPU for Theano (all code here executed in a completely 'fresh' shell):

```
Python 3.4.4 (default, Dec 12 2016, 17:13:14) on linux (64 bits).
This is the Pyzo interpreter with integrated event loop for PYSIDE.
Type 'help' for help, type '?' for a list of *magic* commands.
>>> import theano
>>> import pymc3
>>> # all good, because we're running with theano.device=cpu
```

as does this (note the import of pymc3 before theano):

```
Python 3.4.4 (default, Dec 12 2016, 17:13:14) on linux (64 bits).
This is the Pyzo interpreter with integrated event loop for PYSIDE.
Type 'help' for help, type '?' for a list of *magic* commands.
>>> import pymc3
Using gpu device 0: TITAN X (Pascal) (CNMeM is disabled, cuDNN 5105)
>>> import theano
>>> # also all good
```

but the following results in an error:

```
Python 3.4.4 (default, Dec 12 2016, 17:13:14) on linux (64 bits).
This is the Pyzo interpreter with integrated event loop for PYSIDE.
Type 'help' for help, type '?' for a list of *magic* commands.

>>> import theano
Using gpu device 0: TITAN X (Pascal) (CNMeM is disabled, cuDNN 5105)

>>> import pymc3
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/eelke/repos/pymc3/pymc3/__init__.py", line 14, in <module>
    from . import sampling
ImportError: cannot import name 'sampling'
```

**Comment by twiecki** [comment]
Super odd... anyone have ideas?

**Comment by springcoil** [comment]
What versions of Theano and PyMC3 - how where they installed? Seems to be some sort of sys.path error but this is bizarre...

**Comment by Spaak** [comment]
Sorry for the slow reply. These are the latest master branches from github (just checked, issue persists), installed within a Python virtualenv using `python setup.py develop`.

**Comment by twiecki** [comment]
@nouiz Any idea what this could be?

I wonder if perhaps the test_value could have an influence here.

**Comment by nouiz** [comment]
Strange. I would guess more a problem of import order that cause a circular import. Can you give us the python stack trace when the error happen?

Do you have the time to test the new gpu back-end of Theano? Maybe it don't have this problem.

**Comment by springcoil** [comment]
@Spaak Is this still an issue?

**Comment by twiecki** [comment]
@Spaak Assuming this is fixed, seems like #2138 provides a fix by upgrading gpuarray.
